src/coreclr/pal/src/CMakeLists.txt

```
@@ -318,6 +318,11 @@ if(CLR_CMAKE_TARGET_LINUX)
     target_link_libraries(coreclrpal ${UNWIND_LIBS})
   endif(CLR_CMAKE_USE_SYSTEM_LIBUNWIND)
 
+  # bundled libunwind requires using libucontext on alpine and x86
+  if(CLR_CMAKE_HOST_ALPINE_LINUX AND CLR_CMAKE_HOST_ARCH_I386)
+    add_linker_flag(-lucontext)
+  endif(CLR_CMAKE_HOST_ALPINE_LINUX AND CLR_CMAKE_HOST_ARCH_I386)
```

We are using `TARGET` in this scope for cross compilation:
```suggestion
  if(CLR_CMAKE_TARGET_ALPINE_LINUX AND CLR_CMAKE_TARGET_ARCH_I386)
    target_link_libraries(coreclrpal ucontext)
  endif(CLR_CMAKE_TARGET_ALPINE_LINUX AND CLR_CMAKE_TARGET_ARCH_I386)
```

(note that coreclr, libs and corehost have different host / target semantics than mono; mono uses autoconf semantics {host,build,target} as opposed to {host,target}, which is why we see HOST under mono most of the times and TARGET in rest of the repo)